pennylane/gradients/hybrid_pulse_gradient.py

```diff
@@ -0,0 +1,564 @@
+
+        .. math::
+
+            \frac{\partial U(\{\boldsymbol{\theta_k}\}, [t_0, t_1])}{\partial \theta_{k, r}}
+
+        classically via automatic differentiation, where :math:`\theta_{k, r}` is
+        the :math:`r`\ -th scalar parameter in :math:`\boldsymbol{\theta_k}`.
+
+        Now, due to the compactness of the groups :math:`\mathrm{SU}(N)` , we know that
+        for each :math:`\theta_{k, r}` there is an *effective generator* :math:`\Omega_{k, r}`
+        such that
+
+        .. math::
+
+            \frac{\partial U(\{\boldsymbol{\theta_k}\}, [t_0, t_1])}{\partial \theta_{k,r}} =
+            U(\{\boldsymbol{\theta_k}\}, [t_0, t_1])
+            \frac{\mathrm{d}}{\mathrm{d} x} \exp[x\Omega_{k,r}]\large|_{x=0}.
+
+        Given that we can compute the left-hand side expression as well as the matrix
+        for :math:`U` itself, we can compute :math:`\Omega_{k,r}` for all parameters
+        :math:`\theta_{k,r}`.
+        In addition, we may decompose these generators into Pauli words:
+
+        .. math::
+
+            \Omega_{k,r} =\sum_{\ell=1}^{L} \omega_{k,r}^{(\ell)} \left(P_{\ell}\right)
+
+        The coefficients :math:`\omega_{k,r}^{(\ell)}` of the generators can be computed
+        by decomposing the anti-Hermitian matrix :math:`\Omega_{k,r}` into the Pauli
+        basis and only keeping the non-vanishing terms. This is possible by a tensor
+        contraction with the full Pauli basis (or alternative, more efficient methods):
+
+        .. math::
+
+            \omega_{k,r}^{(\ell)} = \frac{1}{2^N}\mathrm{Tr}\left[P_\ell \Omega_{k,r}\right]
+
+        where :math:`N` is the number of qubits.
+        We may use all of the above to rewrite the partial derivative of an expectation
+        value-based objective function:
+
+        .. math::
+
+            C(\{\boldsymbol{\theta_k}\}, [t_0, t_1])&=
+            \langle\psi_0|U(\{\boldsymbol{\theta_k}\}, [t_0, t_1])^\dagger B
+            U(\{\boldsymbol{\theta_k}\}, [t_0, t_1]) |\psi_0\rangle\\
+            \frac{\partial C}{\partial \theta_{k,r}} (\{\boldsymbol{\theta_k}\}, [t_0, t_1])&=
+            \langle\psi_0|\left[U^\dagger B U, \Omega_{k,r}\right]|\psi_0\rangle\\
+            &=\sum_{\ell=1}^L \omega_{k,r}^{(\ell)}
+            \langle\psi_0|\left[U^\dagger B U, P_\ell \right]|\psi_0\rangle\\
+            &=\sum_{\ell=1}^L \tilde\omega_{k,r}^{(\ell)}
+            \langle\psi_0|\left[U^\dagger B U, -\frac{i}{2}P_\ell \right]|\psi_0\rangle
+            &=\sum_{\ell=1}^L \tilde\omega_{k,r}^{(\ell)}
+            \frac{\mathrm{d}}{\mathrm{d}x} C_\ell(x)\large|_{x=0}
+
+        where we skipped the arguments of the unitary for readability, introduced the
+        modified coefficients :math:`\tilde\omega_{k,r}^{(\ell)}=2i\omega_{k,r}^{(\ell)}`,
+        and denoted by :math:`C_\ell(x)` the modified cost function that has a Pauli
+        rotation about :math:`-\frac{x}{2}P_\ell` inserted before the parametrized time
+        evolution.
```

Comment:
I think this could be expanded. In terms of space, we are already quite deep and elaborate, so might as well be explicit here :)

Reply:
Gave it a shot, let me know whether you'd like to have additional information/explainers in there :)